src/common/histogram.hpp

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "metric_description.hpp"
+#include "registry.hpp"
+
+namespace shar::metrics {
+
+using HistogramFamily = prometheus::Family<prometheus::Histogram>;
+
+struct HistogramRemover {
+  HistogramFamily* m_family;
+  void operator()(prometheus::Histogram* histogram) {
+    m_family->Remove(histogram);
+  }
+};
+
+using HistogramPtr = std::unique_ptr<prometheus::Histogram, HistogramRemover>;
+
+class Histogram {
+
+public:
+  Histogram();
```

Comment:
Could be declared as `= default` if in-class initializer is used for `m_histogram`